Cron/EventsTopic.php

```diff
@@ -0,0 +1,190 @@
+            }
+        }
+
+        // Update Status of rows in kl_events table to Moved
+        $eventsCollection->updateRowStatus($idsMoved, 'MOVED');
+    }
+
+    /**
+     * Deletes rows moved to the kl_syncs table that are older than 2 days
+     */
+    public function deleteMovedRows()
+    {
+        // Delete rows that have been moved to sync table
+        $eventsCollection = $this->_eventsCollectionFactory->create();
+        $idsToDelete = $eventsCollection->getIdsToDelete('MOVED');
+
+        $eventsCollection->deleteRows($idsToDelete);
+    }
+
+    /**
+     * Helper function to replace QuoteId with MaskedQuoteId and CategoryIds with CategoryNames in Added To Cart payload
+     * @param string $payload
+     * @return array
+     */
+    public function replaceQuoteIdAndCategoryIds(string $payload): array
+    {
+        $decoded_payload = json_decode($payload, true);
+        $maskedQuoteId = $this->_quoteIdMaskResource->getMaskedQuoteId(($decoded_payload['QuoteId']));
+        $decoded_payload['MaskedQuoteId'] = $maskedQuoteId;
+        unset($decoded_payload['QuoteId']);
+
+        // Replace CategoryIds for Added Item, Quote Items with resp. CategoryNames
+        $decoded_payload = $this->replaceCategoryIdsWithNames($decoded_payload);
+
+        return $decoded_payload;
+    }
+
+    /**
+     * Replace all CategoryIds in event payload with their respective names
+     * @param $payload
+     * @return array
+     */
+    public function replaceCategoryIdsWithNames(array $payload): array
+    {
+        $categoryMap = $this->getCategoryMap($payload['Categories']);
+
+        foreach ($payload['Items'] as &$item){
+            $item['Categories'] = $this->searchCategoryMapAndReturnNames($item['Categories'], $categoryMap);
+        }
+
+        $payload['AddedItemCategories'] = $this->searchCategoryMapAndReturnNames($payload['AddedItemCategories'], $categoryMap);
+        $payload['Categories'] = array_values($categoryMap);
+
+        return $payload;
+    }
+
+    /**
+     * Retrieve categoryNames from category factory using Ids
+     * @param array $categoryIds
+     * @return array
+     */
+    public function getCategoryMap(array $categoryIds): array
+    {
+        $categoryFactory = $this->_categoryFactory->create();
+        $categoryMap = [];
+
+        foreach($categoryIds as $categoryId){
+            $categoryMap[$categoryId] = $categoryFactory->load($categoryId)->getName();
+        }
+
+        return $categoryMap;
+    }
+
+    /**
+     * Return array of CategoryNames from CategoryMap using ids
+     * @param array $categoryIds
+     * @param array $categoryMap
+     * @return array
+     */
+    public function searchCategoryMapAndReturnNames(array $categoryIds, array $categoryMap): array
+    {
+        $categoryNames = [];
+        foreach ($categoryIds as $id){
+            array_push($categoryNames, $categoryMap[$id]);
+        }
+
+        return $categoryNames;
```

Comment:
```suggestion    
        return array_filter($categoryIds, function($id) {
            return $categoryMap[$id];
        });
```

Reply:
It looks like not an equal to what was there before. I guess you wanted to write here array_map? :)

Reply:
yea my code above fails, I guess I was just suggesting if you can find a lambda way of getting the end result

Reply:
Found a better way to achieve this, coming up in next commit